pkg/ccl/sqlproxyccl/proxy_handler.go

```diff
@@ -451,6 +476,61 @@ func (handler *proxyHandler) handle(ctx context.Context, incomingConn net.Conn)
 	}
 }
 
+// handleCancelRequest handles a pgwire query cancel request by either
+// forwarding it to a SQL node or to another proxy.
+func (handler *proxyHandler) handleCancelRequest(cr *proxyCancelRequest, allowForward bool) error {
+	const timeout = 2 * time.Second
+	if ci, ok := handler.cancelInfoMap.getCancelInfo(cr.SecretKey); ok {
+		if !ci.clientAddr.(*net.TCPAddr).IP.Equal(cr.ClientIP) {
```

Comment:
Can we avoid the cast here by storing a net.TCPAddr on the cancel info?

Reply:
done